gateway/reverse_proxy.go

```diff
@@ -1044,11 +1027,32 @@ func (p *ReverseProxy) WrappedServeHTTP(rw http.ResponseWriter, req *http.Reques
 	}
 
 	p.TykAPISpec.Lock()
+
+	// create HTTP transport
+	createTransport := p.TykAPISpec.HTTPTransport == nil
+
+	// Check if timeouts are set for this endpoint
+	if !createTransport && config.Global().MaxConnTime != 0 {
+		createTransport = time.Since(p.TykAPISpec.HTTPTransportCreated) > time.Duration(config.Global().MaxConnTime)*time.Second
+	}
+
+	if createTransport {
+		_, timeout := p.CheckHardTimeoutEnforced(p.TykAPISpec, req)
+		p.TykAPISpec.HTTPTransport = httpTransport(timeout, rw, req, outreq, p)
+		p.TykAPISpec.HTTPTransportCreated = time.Now()
+	}
+
+	roundTripper = p.TykAPISpec.HTTPTransport
+
 	if roundTripper.transport != nil {
 		roundTripper.transport.TLSClientConfig.Certificates = tlsCertificates
 	}
 	p.TykAPISpec.Unlock()
 
+	if outreq.URL.Scheme == "h2c" {
```

Comment:
We still need to replace it with proper protocol which http stack will undertand.